packages/date-picker/src/shared/components/DateInput/DateInputBox/DateInputBox.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { FocusEventHandler } from 'react';
+import { isSameDay } from 'date-fns';
+import isEqual from 'lodash/isEqual';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useForwardedRef } from '@leafygreen-ui/hooks';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+
+import {
+  DateSegmentsState,
+  isDateSegment,
+  useDateSegments,
+} from '../../../hooks';
+import {
+  doesSomeSegmentExist,
+  getValueFormatter,
+  newDateFromSegments,
+} from '../../../utils';
+import { useDatePickerContext } from '../../DatePickerContext';
+import { DateInputSegment } from '../DateInputSegment';
+import { DateInputSegmentChangeEventHandler } from '../DateInputSegment/DateInputSegment.types';
+
+import {
+  segmentPartsWrapperStyles,
+  separatorLiteralDisabledStyles,
+  separatorLiteralStyles,
+} from './DateInputBox.styles';
+import { DateInputBoxProps } from './DateInputBox.types';
+
+/**
+ * Renders a styled date input with appropriate segment order & separator characters.
+ *
+ * Depends on {@link DateInputSegment}
+ *
+ * Uses parameters `value` & `dateFormat` along with {@link Intl.DateTimeFormat.prototype.formatToParts}
+ * to determine the segment order and separator characters.
+ *
+ * Provided value is assumed to be UTC.
+ *
+ * Argument passed into `setValue` callback is also in UTC
+ * @internal
+ */
+export const DateInputBox = React.forwardRef<HTMLDivElement, DateInputBoxProps>(
+  (
+    {
+      value: dateValue,
+      setValue: setDateValue,
+      className,
+      labelledBy,
+      segmentRefs,
+      onChange: onSegmentChangeProp,
+      ...rest
+    }: DateInputBoxProps,
+    fwdRef,
+  ) => {
+    const { formatParts, disabled } = useDatePickerContext();
+    const { theme } = useDarkMode();
+
+    const containerRef = useForwardedRef(fwdRef, null);
+
+    /**
+     * When a segment is updated,
+     * trigger a `change` event for the segment, and
+     * update the external Date value if necessary
+     */
+    const onSegmentsUpdate = (
+      newSegments: DateSegmentsState,
+      prevSegments?: DateSegmentsState,
+    ) => {
+      const hasAnySegmentChanged = !isEqual(newSegments, prevSegments);
+
+      if (hasAnySegmentChanged) {
+        const utcDate = newDateFromSegments(newSegments);
+        const areAllSegmentsEmpty = !doesSomeSegmentExist(newSegments);
+
+        if (utcDate) {
+          // Only update the value iff all parts are set, and create a valid date.
```

Comment:
nit: `if`

Reply:
"iff" is correct. [If and only if](https://en.wikipedia.org/wiki/If_and_only_if)

Update the value _if_ all parts are set & valid
 - does not exclude the possibility of other cases where the value can be updated
Update the value _only if_ all parts are set & valid
 - does not exclude the possibility of _not_ updating the value when all parts are set & valid

Reply:
I also thought it was `if`, maybe you should add more comments on this?